### PR TITLE
chore(refactor): rename parser to translator everywhere

### PR DESCRIPTION
**Pull request by programmer04**
**What this PR does / why we need it**:

What in the codebase is called `parser` should be rather called `translator` this PR introduces such rename across the codebase. 

**Special notes for your reviewer**:

It was raised during one of the team syncs. The PR is pretty big. It would be understandable to decide not to pursue this change if we decide that the drawbacks outweigh the benefits and live with the old name to the end.

### CODE REVIEW

File: internal/admission/validation/gateway/httproute.go [hunk 1]
```diff
@@ -7,8 +7,8 @@ import (
 
 	"github.com/kong/go-kong/kong"
 
-	"github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/parser"
-	"github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/parser/translators"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/translator"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/translator/translators"
```
Comment (randmonkey) [inline internal/admission/validation/gateway/httproute.go:11]: `translator/translators` looks a bit confusing. Would there be another renaming to give it a better name? (likely to happen in another PR)

**Review by randmonkey — COMMENTED** [review]
LGTM in general, only one discussion left.

**Review by pangruoran — APPROVED** [review]

File: internal/admission/validation/gateway/httproute.go (hunk 1 above)
Comment (programmer04) [reply internal/admission/validation/gateway/httproute.go:11]: good point, I'll address it in separate PR

Comment (czeslavo) [reply internal/admission/validation/gateway/httproute.go:11]: That could be `subtranslators` maybe?

**Review by czeslavo — APPROVED** [review]
I'll miss you, `Parser`. 👋

**Pull request merged**